lib/jsdom/living/xmlhttprequest.js

```diff
@@ -905,7 +919,9 @@ module.exports = function createXMLHttpRequest(window) {
       progressObj.loaded = 0;
       progressObj.lengthComputable = true;
     }
-    properties.responseBuffer = new Buffer(0);
+    // properties.responseBuffer = new Buffer(0);
+    // pre-allocate buffer.
+    properties.responseBuffer = Buffer.alloc(properties.bufferStepSize);
```

Comment:
Since this uses Buffer.alloc, it will have to wait until v10.0.0, which is when we drop Node.js v4 support. That is fine; my next jsdom release day will release v10.0.0.